crates/relayer/src/link/relay_path.rs

```diff
@@ -1381,11 +1389,27 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> RelayPath<ChainA, ChainB> {
         dst_info: &ChainStatus,
         height: Height,
     ) -> Result<(Option<Any>, Option<Any>), LinkError> {
+        crate::time!(
+            "build_recv_or_timeout_from_send_packet_event",
+            {
+                "src_channel": event.packet.source_channel,
+                "dst_channel": event.packet.destination_channel,
+            }
+        );
         let timeout = self.build_timeout_from_send_packet_event(event, dst_info)?;
         if timeout.is_some() {
             Ok((None, timeout))
-        } else {
+        } else if event
+            .packet
+            .are_fields_valid(self.max_memo_size, self.max_receiver_size)
+        {
             Ok((self.build_recv_packet(&event.packet, height)?, None))
+        } else {
+            debug!(
+                "memo and/or receiver field are too big for packet {:#?}",
+                event.packet
+            );
```

Comment:
Rather than outputting the whole packet here (which may be extremely large and flood the console), let's instead log the memo and receiver sizes and the max allowed sizes.

For this you might have to change the validation function again to return both sizes.

Reply:
We can perhaps log the whole packet with `trace!`. What do you think @ljoss17  @ancazamfir?

Reply:
I wanted to keep the validation method as simple as possible, and since it's a debug log I would be inclined to have more details in the logs. I could also put it as trace.

Or would you still prefer to have only the field sizes in the log outputs?

Reply:
it would be nice to have the sequence number and field sizes in the debug log when we choose to not relay the packet. Will surely help debugging along the line.

Let's dump the whole packet data in `trace!` mode.

Reply:
I introduced more advanced logs in this commit 9af4e6d0688748f09be91d09b22c7f272e82fb8c.
With debug logs when one or both fields fails and trace logs of the packet when there is any failure.